src/docker-in-docker/install.sh

```diff
@@ -332,13 +333,64 @@ fi
+    latest_version="$1"
+    repo_url=$2
+    # curl -s "$repo_url" | jq -r --arg latest "$latest_version" '.[].tag_name | select(. != $latest)' | sort -rV | head -n 1   # this helps filter tag_names based on semver specifications, this one was incorrectly installing 0.12.0-rc2 which is the third in the list i.e seeing from top to bottom
+    curl -s "$repo_url" | jq -r 'del(.[].assets) | .[1].tag_name' # this would del the assets key and then get the second encountered tag_name's value from the filtered array of objects
+}
+
+# Function to change the patch number in a semver version
+change_patch_number() {
+    local version="$1"  # Input version
+    local new_patch="$2"  # New patch number
+    # Extract major, minor, and current patch numbers
+    local major=$(echo "$version" | cut -d. -f1)
+    local minor=$(echo "$version" | cut -d. -f2)
+    local current_patch=$(echo "$version" | cut -d. -f3)
+    # Construct the new version with the updated patch number
+    local new_version="$major.$minor.$new_patch"
+    echo "$new_version"
+}
+
 if [ "${INSTALL_DOCKER_BUILDX}" = "true" ]; then
     buildx_version="latest"
     find_version_from_git_tags buildx_version "https://github.com/docker/buildx" "refs/tags/v"
-
     echo "(*) Installing buildx ${buildx_version}..."
     buildx_file_name="buildx-v${buildx_version}.linux-${architecture}"
-    cd /tmp && wget "https://github.com/docker/buildx/releases/download/v${buildx_version}/${buildx_file_name}"
+    if [ $FALLBACK = "true" ]; then # flag to initiate testing of fallback feature
+        set -x # for getting detailed logs in case of a failure - flag to mark start of code
+        new_patch_number="42" # for testing a tag not found scenario for docker/buildx plugin
+        buildx_version_fallback_test=$(change_patch_number "$buildx_version" "$new_patch_number") # for testing a tag not found scenario for docker/buildx plugin
+        buildx_file_name="buildx-v${buildx_version_fallback_test}.linux-${architecture}"
+        buildx_version=$buildx_version_fallback_test
+    fi 
+    cd /tmp
+    wget https://github.com/docker/buildx/releases/download/v${buildx_version}/${buildx_file_name} || {
+        wget_exit_code=$?
+        if [ $wget_exit_code -ne 0 ]; then # means wget command to fetch latest version failed
+            if [ $wget_exit_code -eq 8 ]; then  # failure due to 404: Not Found.
+                echo "wget command failed with HTTP error 404 Not Found"
+                repo_url="https://api.github.com/repos/docker/buildx/releases" # GitHub repository URL
+                latest_version=$(get_latest_version "${repo_url}")  # can take latest_version from fn get_latest_version
+                # latest_version="v${buildx_version}" # can also take latest_version from fn find_version_from_git_tags
+                previous_version=$(get_previous_version "${latest_version}" "${repo_url}")
+                buildx_file_name="buildx-${previous_version}.linux-${architecture}"
```

Comment:
Honestly I would delete all the existing `find_version_from_git_tags` stuff, because it's unreliable for finding releases, and only keep the `get_latest_version()` which you introduced.

Reply:
> Honestly I would delete all the existing `find_version_from_git_tags` stuff, because it's unreliable for finding releases, and only keep the `get_latest_version()` which you introduced.

The GitHub API has strict number of attempts limits and may fail.
I think that's why we are using Git tags here.

Reply:
Okay, I see, what if instead, first try to use the releases API, and if a rate limit is reached, fall back to git tags?

Reply:
> The GitHub API has strict number of attempts limits and may fail.
> I think that's why we are using Git tags here.

Yes, we should only attempt to use GitHub APIs if it is necessary This way we will avoid hitting the limits for users. Folks might require it for other purposes, so we shouldn't consume them as we have another alternative.

Hence, the approach in this PR makes sense to me. Find latest version with `find_version_from_git_tags`, if that version is currently in the release process (so it would fail to download artifacts), then attempt to use the API.

Reply:
thanks 👍 have corrected.